src/Language/PureScript/Renamer.hs

```diff
@@ -76,19 +64,33 @@ newScope x = do
 -- unique name is generated and stored.
 --
 updateScope :: Ident -> Rename Ident
-updateScope i@(Ident name) | name == C.__unused = return i
-updateScope name = do
-  scope <- get
-  name' <- if name `S.member` rsUsedNames scope
-             then do
-               let newNames = [ Ident (runIdent name ++ "_" ++ show (i :: Int)) | i <- [1..] ]
-                   Just newName = find (`S.notMember` rsUsedNames scope) newNames
-               return newName
-             else return name
-  modify $ \s -> s { rsBoundNames = M.insert name name' (rsBoundNames s)
-                   , rsUsedNames  = S.insert name' (rsUsedNames s)
-                   }
-  return name'
+updateScope ident =
+  case ident of
+    Ident name
+      | name == C.__unused -> return ident
+      | last name == '\'' -> do
+          scope <- get
+          go ident $ getNewName (rsUsedNames scope) (Ident $ init name)
+    GenIdent name _ -> go ident $ Ident (fromMaybe "v" name)
```

Comment:
Any particular reason for using `v`?

Didn't you want to try the lettering scheme instead here? `a, b, c, ..., aa, ab, ...` etc.?